### PR TITLE
Add new_test/test target defaultmap none.f90

### PR DESCRIPTION
**Pull request by seyonglee**
Add a new Fortran test benchmark: tests/5.0/target/test_target_defaultmap_none.F90
Tested with both xl/16.1.1-10 and gcc/11.1.0 on Summit, but both failed because defaultmap clause is not supported yet.

### CODE REVIEW

File: tests/5.0/target/test_target_defaultmap_none.F90 [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    DO i = 1, N
+       A(i) = 0
+       B(i) = 0
+       new_struct%SA(i) = 0
+    END DO
+
+    errors = 0
+
+    !$omp target defaultmap(none) map(tofrom: scalar, A, B, new_struct, ptr)
+    scalar = 17 !scalar firstprivate, value not returned
```
Comment (spophale) [inline tests/5.0/target/test_target_defaultmap_none.F90:57]: I think if to from is explicit then the scalar value will be copied back.

**Review by spophale — APPROVED** [review]

File: tests/5.0/target/test_target_defaultmap_none.F90 (hunk 1 above)
Comment (seyonglee) [reply tests/5.0/target/test_target_defaultmap_none.F90:57]: The comment seems to be wrong; what I wanted to say was that scalar variables without TARGET, ALLOCATABLE, or POINTER attribute have an implicit data-sharing attribute of first-private. (OpenMP V5.0 page 315 line 9 ~ 12). Is it correct interpretation?

Comment (tmh97) [reply tests/5.0/target/test_target_defaultmap_none.F90:57]: @seyonglee IMO that is the correct interpretation, I see what you were implying with your comment. @spophale This looks good-to-go. I will approve and leave it up for you to merge. Thanks again @seyonglee !

**Comment by tmh97** [comment]
I removed the defaultmap(none) and ran with nvhpc and gcc, passes on nvfortran but fails on gcc/11.1.10 with out-of-memory issue

File: tests/5.0/target/test_target_defaultmap_none.F90 (hunk 1 above)
Comment (seyonglee) [reply tests/5.0/target/test_target_defaultmap_none.F90]: Changed the comments for the scalar variable to clarify the intention.
